code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -93,6 +93,15 @@
 				usr.equip_to_slot_if_possible(src, slot_l_hand)
 		src.add_fingerprint(usr)
 
+/obj/item/storage/AltClick(var/mob/usr)
+	if(!canremove)
+		return ..()
+	if ((ishuman(usr) || isrobot(usr) || issmall(usr)) && !usr.incapacitated() && Adjacent(usr))
```

Comment:
i think you should be able to replace all of this with if(!use_check_and_message(usr)), would look a lot nicer